cloudinit/config/cc_ca_certs.py

```diff
@@ -162,20 +169,33 @@ def handle(name, cfg, cloud, log, _args):
     @param log: Pre-initialized Python logger object to use for logging.
     @param args: Any module arguments from cloud.cfg
     """
-    # If there isn't a ca-certs section in the configuration don't do anything
-    if "ca-certs" not in cfg:
+    if "ca-certs" in cfg:
+        log.warning(
+            "DEPRECATION: key 'ca-certs' is now deprecated. Use 'ca_certs' by"
+            " version 23.1."
+        )
+        ca_cert_cfg = cfg["ca-certs"]
+    elif "ca_certs" not in cfg:
         log.debug(
             "Skipping module named %s, no 'ca-certs' key in configuration",
             name,
         )
         return
-
-    ca_cert_cfg = cfg["ca-certs"]
+    else:
```

Comment:
There's that small possibility they provide both `ca-certs` and `ca_certs`. In that case I think we should log a warning and use `ca_certs`.

Reply:
+1 added warning and unit test